hrms/hr/doctype/leave_application/leave_application.py

```diff
@@ -791,6 +794,16 @@ def create_ledger_entry_for_intermediate_allocation_expiry(self, expiry_date, su
 				args.update(dict(from_date=start_date, to_date=self.to_date, leaves=leaves * -1))
 				create_leave_ledger_entry(self, args, submit)
 
+	def validate_for_self_approval(self):
+		self_leave_approval_allowed = frappe.db.get_single_value("HR Settings", "allow_self_leave_approval")
+
+		if (not self_leave_approval_allowed) and (
+			self.employee == get_current_employee_info().get("name")
+			if get_current_employee_info()
+			else None and not get_workflow_name("Leave Application")
+		):
+			frappe.throw(_("Self-approval for leaves is not allowed"), frappe.ValidationError)
```

Comment:
```suggestion
		employee_user = frappe.db.get_value("Employee", self.employee, "user_id")
		if (
			employee_user == frappe.session.user 
			and not self_leave_approval_allowed 
			and not get_workflow_name("Leave Application")
		):
			frappe.throw(_("Self approval for leaves is not allowed"))
```
- Simplify checks
- `get_current_employee_info` fetches additional columns that aren't required here. We only need to compare the session user.
- Drop frappe.ValidationError, that's the default exception
![image](https://github.com/user-attachments/assets/7dd89137-5b58-4ea7-af3f-4e49816ef981)